server/admin/admin_server_test.go

```diff
@@ -260,24 +277,35 @@ func TestServer_CreateTree(t *testing.T) {
 	defer ctrl.Finish()
 
 	// PEM on the testonly trees is ECDSA, so let's use an ECDSA key for tests.
-	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
 	if err != nil {
 		t.Fatalf("Error generating test key: %v", err)
 	}
 
-	invalidTree := *testonly.LogTree
+	publicKeyDER, err := x509.MarshalPKIXPublicKey(privateKey.Public())
+	if err != nil {
+		t.Fatalf("Error marshaling public key: %v", err)
+	}
+
+	// Need to change the public key to correspond with the private key generated above.
+	validTree := *testonly.LogTree
+	validTree.PublicKey = &trillian.PublicKey{
+		Der: publicKeyDER,
+	}
+
```

Comment:
Add a test for a tree that doesn't have the PubKey set, and check that the returned key is correct?